src/utils/isDerivedReactComponent.js

```diff
@@ -0,0 +1,22 @@
+// @flow
+import { Component } from 'react'
+
+const cachedResults = new Map()
```

Comment:
browser support isn't awesome on map, this will probably break old browsers—need to fall back to no caching in old browsers most likely?

Reply:
It's required for React 16, so no big deal.

Reply:
Oh shit it is? Are you sure? That'd be huge, them dropping IE 9 and 10!

Reply:
Well, just requiring polyfills anyway 😛

Reply:
@probablyup You probably would use a `WeakMap`. I've seen it's used for attrs which sticks around since it's never not present (and thus GC'd), but just to be sure I'd still use it :)

Reply:
So only `Map` and `Set` are guaranteed to be polyfilled due to React 16's environment requirements. If we wanted to use a `WeakMap` we'd have to ship a polyfill for it and bloat the library size.